bootstrap-obvious-ci-and-miniconda.py

```diff
@@ -85,22 +84,22 @@ def main(
     else:
         raise ValueError("Unsupported operating system.")
 
-    if not os.path.exists(basename):
+    if not Path(basename).exists():
         print("Downloading from {}".format(URL))
         urlretrieve(URL, basename)
     else:
         print("Using cached version of {}".format(URL))
 
     # Install with powershell.
-    if os.path.exists(target_dir):
+    if Path(target_dir).exists():
         raise IOError("Installation directory already exists")
     subprocess.check_call(cmd)
 
-    if not os.path.isdir(target_dir):
+    if not Path(target_dir).is_dir():
         raise RuntimeError("Failed to install miniconda :(")
 
     if install_obvci:
-        conda_path = os.path.join(target_dir, bin_dir, "conda")
+        conda_path = Path(target_dir, bin_dir, "conda")
         subprocess.check_call(
```

Comment:
I don't understand why we are using several `Path` instantiations here when it's all `target_dir`. We should have something like:

```
target_dir = Path(target_dir)
if target_dir.exists():
  ...
if not target_dir.is_dir():
  ...
if install_obvci:
  conda_path = target_dir / bin_dir / "conda"
```

and so on. I'm afraid this happens in more areas of the PR, so a careful review is advised imo.